snapcraft/internal/repo/_deb.py

```diff
@@ -331,13 +331,15 @@ def _install_new_build_packages(cls, package_names: List[str]) -> None:
             {"DEBIAN_FRONTEND": "noninteractive", "DEBCONF_NONINTERACTIVE_SEEN": "true"}
         )
 
-        apt_command = ["sudo", "apt-get", "--no-install-recommends", "-y"]
+        apt_command = ["sudo", "apt-get", "-y"]
         if not is_dumb_terminal():
             apt_command.extend(["-o", "Dpkg::Progress-Fancy=1"])
-        apt_command.append("install")
+        apt_update = apt_command + ["update"]
```

Comment:
if this is mostly for docker, then common.is_docker should be checked for.

Reply:
This is for all sorts of destructive installs, because without `update` `install` will try to download missing files and fail if there is a new version of index.